### PR TITLE
Move benchmark download tool and visualization to the CI

### LINKED ISSUE #8857: Generate benchmark result pages in the CI
Since its inception, the benchmark results have been collected, saved into cache on my local file system, and uploaded to https://github.com/enso-org/engine-benchmark-results. It is scheduled to run as a daily job. The https://github.com/enso-org/enso/blob/49835500895be8b1f779cc2c40df9c03ee3ebcb8/tools/performance/engine-benchmarks/bench_download.py script requires a (local) cache because the GH artifacts are dropped after 3 months (this period cannot be extended, according to the GH policies), so we would lose the benchmark results older than 3 months. The cache is located on my local file system, without any backups.

Let's add another daily GH action that runs `bench_download.py`.

```[tasklist]
### Tasks
- [x] Reimplement the cache from local filesystem to https://github.com/enso-org/engine-benchmark-results
- [ ] Add another daily GH action that runs `bench_download.py`.
- [ ] Generate a GH token (permissions described in https://github.com/enso-org/enso/issues/8857#issuecomment-1914558703)
```

A rather more difficult, but more permanent, solution is to implement just a frontend (hosted as a website in https://github.com/enso-org/engine-benchmark-results repo) that fetches all the bench results dynamically from the GH artifacts. Such a solution would have the following properties:
- Do not collect results older than 3 months into any kind of cache - no need for any kind of backend.
- Benchmark comparison (compare a custom branch to `develop) can be handled by the website
- No more need for a custom `bench_download.py` script - everything could be done on the website.
- The advantage is that we would see all the results immediately.

### PR DESCRIPTION
**Pull request by Akirathan**
Closes #8857

### Pull Request Description

Creates a new [Benchmarks upload](https://github.com/enso-org/enso/pull/9075/files#diff-8859b4f24c2f25d300fe800ee431d7f9f7e68de459395a7e9b22abf79440c862) GitHub action that fetches all the latest benchmark results, and uploads them on the website hosted on https://github.com/enso-org/engine-benchmark-results repo. The results are stored in that repo in a bunch of JSON files.

### Important Notes

The new *Benchmarks upload* action is scheduled to run after either "Engine benchmarks" or "Standard library benchmarks" jobs are complete.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] Revert 7dc9ed703369d097c7247c6039e8cb1237e736a2
- [X] The documentation has been updated, if necessary.
- [X] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [X] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [X] Unit tests have been written where possible.
  - [X] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Comment by Akirathan** [comment]
@mwu-tow It seems that the [ENSO_BENCHMARK_RESULTS_TOKEN](https://github.com/enso-org/enso/issues/8857#issuecomment-1914653799) does not work for pushing to the engine-bnechmark-results repo. See the failure in https://github.com/enso-org/enso/actions/runs/8018190361/job/21903605530?pr=9075#step:6:4678. Am I doing something wrong, or are the permissions wrong for that token? I am setting the GITHUB_TOKEN env var in https://github.com/enso-org/enso/blob/e1ee00d854dcd3f11aeab7d8b903db182b03679f/.github/workflows/bench-upload.yml#L46-L47

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/enso-org/enso">Move benchmark download tool and visualization to the CI · enso-org/enso@e1ee00d</a></strong></div><div>Hybrid visual and textual functional programming. Contribute to enso-org/enso development by creating an account on GitHub.</div></blockquote>

File: tools/performance/engine-benchmarks/README.md
Comment (JaroslavTulach) [inline tools/performance/engine-benchmarks/README.md:1]: I'd like to point out that the `engine-benchmarks` name is no longer fully correct. The tool supports `-s engine` as well as `-s stdlib` - e.g. it is not only for _engine benchmarks_.

**Comment by JaroslavTulach** [comment]
Please add:
```diff
tools/performance/engine-benchmarks$ git diff .
diff --git tools/performance/engine-benchmarks/templates/template_jinja.html tools/performance/engine-benchmarks/templates/template_jinja.html
index 97311c0d8a..9054a00662 100644
--- tools/performance/engine-benchmarks/templates/template_jinja.html
+++ tools/performance/engine-benchmarks/templates/template_jinja.html
@@ -334,7 +334,8 @@
     <div id="benchmarks-container" class="container-fluid">
       {% for bench_data in bench_datas %}
       <div class="bench-container card">
-        <h3 class="card-header text-center">{{ bench_data.id }}</h3>
+        <a name="{{ bench_data.id }}"></a>
+        <h3 class="card-header text-center"><a href="#{{ bench_data.id }}">{{ bench_data.id }}</a></h3>
         <!-- This is a placeholder div for a bench chart -->
         <div id="{{ bench_data.id }}" class="bench-chart"></div>
         <!-- selection-info div will be shown once user selects a point in the chart -->

```
to the template, so people can reference individual benchmarks on the page with an anchor: `stdlib-benchs.html#org_enso_benchmarks_generated_Startup_hello_world_startup`

**Comment by Akirathan** [comment]
> Please add ... to the template, so people can reference individual benchmarks on the page with an anchor: stdlib-benchs.html#org_enso_benchmarks_generated_Startup_hello_world_startup

@JaroslavTulach  Good idea. Done in 7da61c724f0d721ec91150ae66e77dd088347937

**Comment by Akirathan** [comment]
The first successful run (that uploads the website) of the action is in https://github.com/enso-org/enso/actions/runs/8078775389?pr=9075

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/enso-org/enso">Move benchmark download tool and visualization to the CI · enso-org/enso@d7283ae</a></strong></div><div>Hybrid visual and textual functional programming. Contribute to enso-org/enso development by creating an account on GitHub.</div></blockquote>

**Review by mwu-tow — APPROVED** [review]

**Comment by Akirathan** [comment]
After some experimentation, the last successful run is in https://github.com/enso-org/enso/actions/runs/8079939312/job/22075410072?pr=9075. I believe it is time to merge this PR, and remove the `on: pull_request` trigger.

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/enso-org/enso">Move benchmark download tool and visualization to the CI · enso-org/enso@76802d3</a></strong></div><div>Hybrid visual and textual functional programming. Contribute to enso-org/enso development by creating an account on GitHub.</div></blockquote>

**Review by hubertp — APPROVED** [review]

**Pull request merged**